data_structures/stack/stack.c

```diff
@@ -45,17 +45,8 @@ void grow()
 {
     max += 10; /* increases the capacity */
 
-    int i;  // for the loop
-    void **tmp = malloc(sizeof(void *) * max);
-
-    /* copies the elements from the origin array in the new one. */
-    for (i = 0; i < max - 10; i++)
-    {
-        *(tmp + i) = *(array + i);
-    }
-    /*free the memory */
-    free(array);
-    array = tmp;
+    array = (void **)realloc(array, sizeof(void *) * max);
+    assert(array); /* tests whether pointer is assigned to memory. */
```

Comment:
If asserts are disabled and `realloc` would return `NULL`, there will be a memory leak.

[C++ reference on `realloc`](https://en.cppreference.com/w/c/memory/realloc) states
> If there is not enough memory, the old memory block is not freed and null pointer is returned.

Reply:
There is no problem with using assert. I simply said that the suggested code (although much better than the original one) can lead to a very tedious bug (it will not occur often, but when it will occur it will be hard to find).

In any case, the idea of explicitly checking with `array == NULL` looks reasonable.